bpf-gpl/tc.c

```diff
@@ -379,25 +315,6 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		}
 	}
 
-	/* [SMC] I had to add this revalidation when refactoring the conntrack code to use the context and
-	 * adding possible packet pulls in the VXLAN logic.  I believe it is spurious but the verifier is
-	 * not clever enough to spot that we'd have already bailed out if one of the pulls failed. */
-	if (skb_refresh_validate_ptrs(&ctx, UDP_SIZE)) {
-		ctx.fwd.reason = CALI_REASON_SHORT;
-		CALI_DEBUG("Too short\n");
-		goto deny;
-	}
```

Comment:
It's good to remove this if we can, but have you done any analysis or experimenting to check that it is OK?

Reply:
All tests are passing. I also loaded the prog manually. Will do more tests.

Reply:
I reverted the change. It seems to work fine. I just think maybe it's better to make this change separately out of this PR.